File Integrity Operator release notes (#39441)
openshift/openshift-docs: security/file_integrity_operator/file-integrity-operator-release-notes.adoc

//OpenShift File Integrity Operator Release Notes
[id="file-integrity-operator-release-notes"]
= File Integrity Operator release notes
:context: file-integrity-operator-release-notes-v0
include::modules/common-attributes.adoc[]

toc::[]

> The File Integrity Operator lets {product-title} administrators describe the required compliance state of a cluster and provides them with an overview of gaps and ways to remediate them.

Author: Whoops, I left that in on accident. Will fix.

These release notes track the development of the File Integrity Operator in the {product-title}.

For an overview of the File Integrity Operator, see xref:../../security/file_integrity_operator/file-integrity-operator-understanding.adoc#understanding-file-integrity-operator[Understanding the File Integrity Operator].

> [id="file-integrity-operator-release-notes-0-1-21"]

Reviewer: Also, Is this the first version of FIO that is released for OpenShift?

> [id="file-integrity-operator-release-notes-0-1-21"]

Author: Yes, that's right.

This is not the first released version for FIO. 0.1.21 is just the currently released version. The previous one was 0.1.16.

== OpenShift File Integrity Operator 0.1.21

The following advisory is available for the OpenShift File Integrity Operator 0.1.21:

* link:https://access.redhat.com/errata/RHBA-2021:4631[RHBA-2021:4631 OpenShift File Integrity Operator Bug Fix and Enhancement Update]

[id="file-integrity-operator-0-1-21-new-features-and-enhancements"]
=== New features and enhancements

> * There are now informational metrics related to the FileIntegrity scan results and processing. The metrics displayed on the monitoring dashboard on the console are labeled with the prefix of `file_integrity_operator_`.

Author: Done!

> * There are now informational metrics related to the ``FileIntegrity` scan results and processing. The metrics displayed on the monitoring dashboard on the console are labeled with the prefix of `file_integrity_operator_`.

Reviewer 2: ```suggestion
* The metrics related to `FileIntegrity` scan results and processing metrics are displayed on the monitoring dashboard on the web console. The results are labeled with the prefix of `file_integrity_operator_`.
```

Avoid using "there are", "there is" as a verb. This is part of the IBM Style Guide: 

>  there is, there are (verb)
Avoid at the beginning of an independent clause. In most cases rewrite to remove the phrase. For example, instead of "There are three recovery control data sets in the product" write "The product includes three recovery control data sets".

> * There are now informational metrics related to the FileIntegrity scan results and processing. The metrics exposed to the monitoring dashboard on the console are labeled with the prefix of `file_integrity_operator_`.

Author: More or less- I think displayed on is reasonable.

+

> * There is a default PrometheusRule provided in the operator namespace that alerts with a warning if a node has an integrity failure for more than 1 second.

Author: Done!

> * There is a default `PrometheusRule` provided in the operator namespace that alerts with a warning if a node has an integrity failure for more than 1 second.

Reviewer 2: ```suggestion
* If a node has an integrity failure for more than 1 second, the default `PrometheusRule` provided in the operator namespace alerts with a warning.
```

Avoid using "there are", "there is" as a verb. This is part of the IBM Style Guide:

> there is, there are (verb)
> Avoid at the beginning of an independent clause. In most cases rewrite to remove the phrase. For example, instead of "There are three recovery control data sets in the product" write "The product includes three recovery control data sets".

+

> * The following dynamic MCO and CVO related filepaths are excluded from the default AIDE policy to help prevent false positives during node updates:

Author: Done!

> * Some dynamic MCO and CVO related filepaths have been excluded from the default AIDE policy to help prevent false positives during node updates. The excluded paths are:

Author: Sounds good, will fix

 - /etc/machine-config-daemon/currentconfig
 - /etc/pki/ca-trust/extracted/java/cacerts
 - /etc/cvo/updatepayloads
 - /root/.kube
+

> * The AIDE daemon process has stability improvements over v0.1.16, and is more resilient to errors during database re-init.

Author: I've incorporated this, good suggestion.

> * The AIDE daemon process has stability improvements and is more resilient to errors during database re-init.

Reviewer: Are we comparing deltas between an FIO upstream version to this released version (0.1.22)?  The terms "stability improvements" and "more resilient" hint at a delta between versions.

Author: I can update it to say that there are stability improvements since v0.1.16, which was the release prior to the current.

[id="openshift-file-integrity-operator-0-1-21-bug-fixes"]
=== Bug fixes

> * Operator updates from pre-0.1.16 versions now handle the deletion of outdated daemonSets.

Author: What we do is make releases upstream, and then at a different intervals we ship those releases downstream. For FIO the last two shipped releases (for OCP 4.6+) were 0.1.16 and 0.1.21. The operators update between the versions automatically.

> * Operator updates from pre-0.1.16 versions now handle the deletion of outdated daemonSets.

Reviewer: Still a bit confused with this sentence. Trying a rewrite..
Previously, when the Operator was updated automatically, outdated daemon sets were not  removed. With this release, outdated daemon sets are removed during the automatic update.

Are Operators automatically updated or automatic upgraded?

> * Previously, when the Operator was automatically upgraded, outdated daemon sets were not removed. With this release, outdated daemon sets are removed during the automatic upgrade.

Reviewer 2: ```suggestion
* Previously, when the Operator automatically upgraded, outdated daemon sets were not removed. With this release, outdated daemon sets are removed during the automatic upgrade.
```

The use of "was" here is not needed.
